packages/peregrine/lib/talons/AddressBookPage/useAddressBookPage.js

```diff
@@ -56,14 +90,101 @@ export const useAddressBookPage = (props = {}) => {
     }, [isRefetching, setPageLoading]);
 
     const handleAddAddress = useCallback(() => {
-        alert('TODO!');
+        // Used to seed the form when adding a new address.
+        // country_code is needed for the Region component to work properly.
+        // default_shipping set to true when this is the first address being added.
+        const formAddress = {
+            country_code: 'US',
+            default_shipping: customerAddresses.length === 0
+        };
+        formApi.reset();
+        formApi.setValues(formAddress);
```

Comment:
We send these as `initialValues` to the form further down in this file but unfortunately we also have to repeat this each time the dialog is opened in `Add` mode.

`initialValues` is only set once and can't be updated; we need to change `default_shipping` based on the current number of customer addresses, which might change as the user interacts with the page (adding / removing addresses).